### PR TITLE
Revert removal of eoi_entry in #13447

### PR DESCRIPTION
**Pull request by jfehrle**
While not used in the Coq source tree, `eoi_entry` is used in at least one plugin.  See discussion starting [here](https://github.com/coq/coq/pull/13447#issuecomment-735307396).

I didn't re-add `map_entry`.

@herbelin Perhaps you could take this one?

@gares We'd like to get this into 8.13.  The reversion may be pointless if it's not.

### CODE REVIEW

**Comment by gares** [comment]
#13447 is not in 8.13, why should this be?

**Comment by jfehrle** [comment]
Either both should go in or neither.  Hard for me to know what's in 8.13.

**Comment by LasseBlaauwbroek** [comment]
Thanks for the PR @jfehrle. In principle, it looks good to me. I can test it with my plugin but it will probably be next week as I'm rather busy with CUDW+other things.

**Comment by herbelin** [comment]
@LasseBlaauwbroek: Didn't you need only `eoi_entry`?

Anyway, not a big deal to also restore `constr_eoi` and `tactic_eoi` even if you don't need them.

Assigning,  but waiting for feedback before merging.

**Comment by LasseBlaauwbroek** [comment]
I just confirmed a minute ago that my plugin builds against this PR. But not with the new push of 30s ago. Because my plugin does depend on `tactic_eoi` and only indirectly on `eoi_entry`.

**Comment by jfehrle** [comment]
> Anyway, not a big deal to also restore `constr_eoi` and `tactic_eoi` even if you don't need them.

Updated to remove those, too.

Let's give @LasseBlaauwbroek another day or two to respond.  Having him double-check this would be nice but I think not strictly necessary.

**Comment by LasseBlaauwbroek** [comment]
@jfehrle Just in case you missed our cross-posting, let me ping you again :-)

**Comment by jfehrle** [comment]
Try it now.

BTW, once I've commented in a PR, I will be notified when anyone comments in the PR (unless I explicitly disable notifications from that thread).

**Comment by LasseBlaauwbroek** [comment]
Looks good :+1:  (although I would advocate keeping `constr_eoi` as well because it is fairly useful and much more discoverable than `eoi_entry` for people unfamiliar with the parser).

**Comment by retzger77** [comment]
Thank 🧒 for all your hard work 😉🐸😉🙏🎉💪

**Comment by herbelin** [comment]
> Looks good 👍 (although I would advocate keeping constr_eoi as well because it is fairly useful and much more discoverable than eoi_entry for people unfamiliar with the parser).

I feel that @ejgallego was also in the direction of thinking that it was better to keep it (but this was not fully explicit).

**Comment by jfehrle** [comment]
OK, I put that back as well.

**Review by herbelin — APPROVED** [review]
Thanks! Merging tomorrow.

**Comment by herbelin** [comment]
@coqbot: merge now

**Pull request merged**
